shell/platform/linux/fl_key_embedder_responder.h

```diff
@@ -44,11 +54,16 @@ G_DECLARE_FINAL_TYPE(FlKeyEmbedderResponder,
  * the event.
  *
  * Creates a new #FlKeyEmbedderResponder.
+ * @send_key_event: a function that is called on every key event.
+ * @send_key_event_user_data: an opaque pointer that will be sent back as the
+ * last argument of send_key_event, created and managed by the object that holds
+ * FlKeyEmbedderResponder.
  *
  * Returns: a new #FlKeyEmbedderResponder.
  */
 FlKeyEmbedderResponder* fl_key_embedder_responder_new(
-    EmbedderSendKeyEvent send_key_event);
+    EmbedderSendKeyEvent send_key_event,
+    void* send_key_event_user_data);
```

Comment:
Since this isn't a callback it shouldn't really need a `user_data` baton. Typically we pass those only during callback registration.

Reply:
It is a callback. The `FlKeyEmbedderResponder` is created with a callback that is invoked every time it needs to dispatch a key event to the engine.

Reply:
Ah I see -- you're right that's just the user data that goes along with the callback. Not sure why I missed this.